core/store/src/test/java/uk/gov/gchq/gaffer/store/library/AbstractGraphLibraryTest.java

```diff
@@ -0,0 +1,278 @@
+    }
+
+    @Test
+    public void shouldNotAddNullProperties() {
+        // When / Then
+        try {
+            graphLibrary.addProperties(null);
+        } catch (final IllegalArgumentException e) {
+            assertTrue(e.getMessage().contains("Id is invalid: null"));
+        }
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenGraphIdWithDifferentPropertiesExists() {
+        // Given
+        graphLibrary.add(TEST_GRAPH_ID, schema, storeProperties);
+
+        // When / Then
+        try {
+            graphLibrary.add(TEST_GRAPH_ID, schema, storeProperties1);
+            fail("Exception expected");
+        } catch (final Exception e) {
+            assertTrue(e.getMessage().contains("already exists with a different store properties"));
+        }
+    }
+
+    @Test
+    public void shouldNotThrowExceptionWhenGraphIdWithSameSchemaExists() {
+        // Given
+        graphLibrary.add(TEST_GRAPH_ID, schema1, storeProperties);
+
+        final Schema schema1Clone = schema1.clone();
+
+        // When
+        graphLibrary.checkExisting(TEST_GRAPH_ID, schema1Clone, storeProperties);
+
+        // Then - no exceptions
+    }
+
+    @Test
+    public void shouldUpdateWhenGraphIdExists() {
+        // When
+        graphLibrary.addOrUpdate(TEST_GRAPH_ID, schema, storeProperties);
+
+        // Then
+        assertEquals(graphLibrary.getProperties(storeProperties.getId()), storeProperties);
+
+        // When
+        graphLibrary.addOrUpdate(TEST_GRAPH_ID, schema, storeProperties1);
+
+        // Then
+        assertEquals(graphLibrary.getProperties(storeProperties1.getId()), storeProperties1);
+    }
+
+    @Test
+    public void shouldReturnNullWhenPropertyIdIsNotFound() {
+        // When
+        final StoreProperties unknownStoreProperties = graphLibrary.getProperties(TEST_UNKNOWN_PROPERTIES_ID);
+
+        // Then
+        assertNull(unknownStoreProperties);
+    }
+
+    @Test
+    public void shouldReturnNullWhenSchemaIdIsNotFound() {
+        // When
+        final Schema unknownSchema = graphLibrary.getSchema(TEST_UNKNOWN_SCHEMA_ID);
+
+        // Then
+        assertNull(unknownSchema);
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenNewStorePropertiesAreAddedWithSamePropertiesIdAndDifferentProperties() {
+        // Given
+        final StoreProperties tempStoreProperties = new StoreProperties(TEST_PROPERTIES_ID);
+        tempStoreProperties.set("randomKey", "randomValue");
+
+        // When
+        graphLibrary.addProperties(storeProperties);
+
+        // Then
+        try {
+            graphLibrary.addProperties(tempStoreProperties);
+            fail("Exception expected");
+        } catch (final OverwritingException e) {
+            assertTrue(e.getMessage().contains("already exists with a different store properties"));
+        }
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenNewSchemaIsAddedWithSameSchemaIdAndSchema() {
```

Comment:
I think this test is wrong.
1) The schema is not the same for the SchemId in library, so it doesn't match the title of this test, "WithSameSchemaIdAndSchema".

2) If you add a new schema and it is the same as the existing schema an error shouldn't be thrown. I think is should possibly be silently ignore.

Reply:
Ah, this was just meant to say different schema.  It is correct for the exception to be thrown when adding a second Schema with the same ID but different Schema.